File: src/adapter/execution/console-execution-target.ts
import type { ExecutionTarget } from "../../domain/ports/execution-target.ts";
import type { SessionResult } from "../../domain/value-objects/session-result.ts";

export class ConsoleExecutionTarget implements ExecutionTarget {
  async deliver(result: SessionResult): Promise<void> {
    console.log(`\n[Session ${result.sessionId}]`);
    console.log(result.output);
  }
}

File: src/HaaS.Adapters/Execution/ConsoleExecutionTarget.cs
using HaaS.Domain.Ports;
using HaaS.Domain.ValueObjects;

namespace HaaS.Adapters.Execution;

public class ConsoleExecutionTarget : IExecutionTarget
{
    public async Task DeliverAsync(SessionResult result)
    {
        await Console.Out.WriteLineAsync(string.Empty);
        await Console.Out.WriteLineAsync($"[Session {result.SessionId}]");
        await Console.Out.WriteLineAsync(result.Output);
    }
}
